packages/react/src/components/InAppMessaging/hooks/useMessageProps/types.ts

```diff
@@ -0,0 +1,50 @@
+import { MessagePayloadStyle } from '@aws-amplify/ui-react-core';
+import React from 'react';
+
+// import { ImageDimensions } from '../useMessageImage';
+
+/**
+ * Resolved message types
+ */
+export interface MessageComponentStyles {
+  body: React.CSSProperties;
+  closeIconButton: React.CSSProperties;
+  container: React.CSSProperties;
+  header: React.CSSProperties;
+  image: React.CSSProperties;
+  primaryButton: React.CSSProperties;
+  secondaryButton: React.CSSProperties;
+}
+
+export interface UseMessageProps {
+  hasButtons: boolean;
+  hasPrimaryButton: boolean;
+  hasRenderableImage: boolean;
+  hasSecondaryButton: boolean;
+  shouldRenderMessage: boolean;
+  styles: MessageComponentStyles | null;
+}
+
+export interface MessageStyleParams {
```

Comment:
Is this extra interface necessary, or could `getMessageStyles` just take a single argument of type `StyleParams`? I guess the question is, are you expecting to receive add more properties here?

Reply:
The biggest reason I left it this way was because I wanted it to have this type maintain some parity with its React Native counterpart. But, in practice, it actually seems I've stripped the hook down to a point where this type isn't even used 😓 so I will remove it altogether!

Reply:
A large part of doing this was simply to maintain parity with the React Native counterpart. We could scrap it and just inline the `StyleParams` typing in the function arguments but it is probably cleaner and more easily extensible (should we decide we do need to add more arguments especially as we add more complex components e.g. Carousel - even though I don't anticipate it yet).